src/components/Output/index.tsx

```diff
@@ -127,6 +123,35 @@ export default class Output extends Component<Props, State> {
     return !shallowEqual(this.props, nextProps) || !shallowEqual(this.state, nextState);
   }
 
+  @bind
+  private async initializeImage() {
+    await this.setRotation(360)();
+    this.setZoom(1)();
+    this.resetPosition();
+    this.setState({ altBackground: false });
+  }
+
+  @bind
+  private handleIntersect(entries: any) {
+    entries.forEach((entry: any) => {
+      this.setState({ isIntersecting: entry.isIntersecting });
+    });
+  }
+
+  @bind
+  private observeIntersection() {
+    if (!this.pinchZoomLeft || !this.canvasLeft) return;
+
+    const options = {
+      root: this.pinchZoomLeft,
+      rootMargin: '0px',
+      threshold: 0.5,
+    };
+    const observer = new IntersectionObserver(this.handleIntersect, options);
```

Comment:
You’d want a feature detection here and lazy-load a polyfill if `IntersectionObserver` is unavailable.